src/client/components/pages/parts/revisions-table.js

```diff
@@ -28,67 +28,107 @@ const {formatDate} = utilsHelper;
 
 
 function RevisionsTable(props) {
-	const {results} = props;
+	const {results, showRevisionNote, showRevisionEditor, tableHeading} = props;
+
 	const tableCssClasses = 'table table-striped';
 	return (
 
 		<div>
 			<div>
-				<h1 className="text-center">Recent Activity</h1>
+				<h1 className="text-center">{tableHeading}</h1>
 			</div>
 			<hr className="thin"/>
-			<Table
-				responsive
-				className={tableCssClasses}
-			>
-				<thead>
-					<tr>
-						<th className="col-sm-2">Revision ID</th>
-						<th className="col-sm-5">Modified entities</th>
-						<th className="col-sm-3">User</th>
-						<th className="col-sm-2">Date</th>
-
-					</tr>
-				</thead>
-				<tbody>
-					{
-						results.map((revision, i) => (
-							<tr key={i}>
-								<td>
-									<a href={`/revision/${revision.revisionId}`} >
-										{revision.revisionId}
-									</a>
-								</td>
-								<td>
-									{revision.entities.map(entity => (
-										<div key={`${revision.revisionId}-${entity.bbid}`}>
-											<a href={getEntityUrl(entity)} >
-												{genEntityIconHTMLElement(entity.type)}
-												{getEntityLabel(entity)}
-											</a>
-										</div>
-									))}
-								</td>
-								<td>
-									<a href={`/editor/${revision.editor.id}`} >
-										{revision.editor.name}
-									</a>
-								</td>
-								<td>{formatDate(new Date(revision.createdAt), true)}</td>
+			{
+				results.length > 0 ?
+					<Table
+						responsive
+						className={tableCssClasses}
+					>
+						<thead>
+							<tr>
+								<th className="col-sm-2">Revision ID</th>
+								<th className="col-sm-5">Modified entities</th>
+								{
+									showRevisionEditor ?
+										<th className="col-sm-3">User</th> : null
+								}
+								{
+									showRevisionNote ?
+										<th className="col-sm-3">Note</th> : null
+								}
+								<th className="col-sm-2">Date</th>
 							</tr>
-						))
-					}
-				</tbody>
-			</Table>
+						</thead>
+
+						<tbody>
+							{
+								results.map((revision, i) => (
+									<tr key={i}>
+										<td>
+											<a href={`/revision/${revision.revisionId}`} >
+												{revision.revisionId}
+											</a>
+										</td>
+										<td>
+											{revision.entities.map(entity => (
+												<div key={`${revision.revisionId}-${entity.bbid}`}>
+													<a href={getEntityUrl(entity)} >
+														{genEntityIconHTMLElement(entity.type)}
+														{getEntityLabel(entity)}
+													</a>
+												</div>
+											))}
+										</td>
+										{
+											showRevisionEditor ?
+												<td>
+													<a href={`/editor/${revision.editor.id}`} >
+														{revision.editor.name}
+													</a>
+												</td> : null
+										}
+										{
+											showRevisionNote ?
+												<td>
+													{revision.noteContent.map(note => (
+														// eslint-disable-next-line react/jsx-key
+														<div>
```

Comment:
There should really be key here. I see that there aren't any great candidates, considering using the array index doesn't work.
Let's instead find something unique to the note: its id.
We don't have access to it here, so you'll have to change `noteContent` instead to a `notes` array of objects {id, authorId, content} (in `getOrderedRevisionForEditorPage`).
You can then safely use `note.id` as the `key`.

It also makes sense if we want to reuse this mechanism for the API, where I would expect that format.

Reply:
Oh the key is used for API, I didn't know that :P

Reply:
No, I didn't explain well :D Sorry !

The key is for React to keep track of reusable DOM elements (it needs to know what elements need to be updated. [Here's a better explanation](https://blog.arkency.com/2014/10/react-dot-js-and-dynamic-children-why-the-keys-are-important/)).

The format of `revision.notes` is the part that matches what I would expect the API to return.
Currently there is no API endpoint to get editors or revisions, so it's a question of shaping that new getOrderedRevisionForEditorPage function to be easier to reuse.

Reply:
Thankss :D